PcapDotNet/src/PcapDotNet.Packets/Http/HttpRequestDatagram.cs

```diff
@@ -46,6 +46,18 @@ public override ILayer ExtractLayer()
                        Body = Body,
                    };
         }
+        
+        /// <summary>
+        /// An HTTP Request is valid if it contains a method, an URI, and a version.
+        /// </summary>
+        protected override bool CalculateIsValid()

```

Comment:
I suggest creating a separate property for this instead of overriding CalculateIsValid().
The name of the property can be something like "IsValidStart".
The reason is that an HttpDatagram can be the continuation of a previous HttpDatagram and so doesn't have to start with the first line to be valid.